app/models/user.rb

```diff
@@ -92,4 +105,26 @@ def ensure_no_token_scope_loss
   def ensure_last_active_at_presence
     self.last_active_at ||= Time.zone.now
   end
+
+  # Internal: Returns a SimpleBox encryptor for token encryption/decryption.
+  def encryptor
+    token_secret = Rails.application.secrets.token_secret
+    raise "TOKEN_SECRET is not set, please check you .env file" if token_secret.blank?
+    @encryptor ||= RbNaCl::SimpleBox.from_secret_key(token_secret.unpack("m0").first)
+  end
+
+  def encrypt_value(value)
+    return if value.blank?
+
+    ciphertext = encryptor.encrypt(value)
+    hex = ciphertext.bytes.pack("c*").unpack("H*").first
```

Comment:
Hex encoding gives a consistent string (no new lines) base64 had some oddities. All the same this could be:

```
  def encrypt_value(value)
    return if value.blank?
    ciphertext = encryptor.encrypt(value)
    "GH_" + Base64.encode64(ciphertext)
  end
```

Reply:
Thanks for getting this rolling? Any reason to do Base64 here if we get a more consistent string with hex? Is Base64 more secure?

Reply:
No reason except encrypted key size